src/main/java/gojava31.ua.goit/Categories.java

```diff
@@ -0,0 +1,50 @@
+package gojava31.ua.goit;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+@WebServlet("/categories")
+public class Categories extends HttpServlet {
+    private static List<Category> categories = new CopyOnWriteArrayList<Category>();
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+		List<Project> projects = new CopyOnWriteArrayList<Project>();
+		projects.add(new Project("1"));
+        projects.add(new Project("2"));
+        addCategory("IT", projects);
+        addCategory("Music", projects);
+        addCategory("Films", projects);
+     }
+
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        response.setContentType("text/html");
+        PrintWriter writer = response.getWriter();
+        String uri = request.getRequestURI();
+        for (Category category : categories) {
+            writer.write(String.format("<a href=\"/viewcategory?category=%s\">%s</a><br>", category.getName(), category.getName()));
+        }
+    }
+
+    private static void addCategory(String name, List<Project> projects) {
+        categories.add(new Category(name, projects));
+    }
+
+    public static Category getCategory(String name) {
```

Comment:
also (just a theoretical one) suggest me another data structure to put categories into which will make lookup by name here faster